clusterloader2/pkg/measurement/util/histogram_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/prometheus/common/model"
+)
+
+func Equal(a, b *HistogramVec) bool {
+	if len(*a) != len(*b) {
+		return false
+	}
+	for i, v := range *a {
+		if !reflect.DeepEqual(v, (*b)[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+func TestConvertSampleToBucket(t *testing.T) {
+	tests := map[string]struct {
+		samples           []*model.Sample
+		inputHistogramVec *HistogramVec
+		want              *HistogramVec
+	}{
+		"convert single sample to bucket": {
+			samples: []*model.Sample{
+				{
+					Metric: model.Metric{"name": "value", "le": "le-value"},
+					Value:  1,
+				},
+			},
+			inputHistogramVec: &HistogramVec{
+				{
+					Labels: map[string]string{
+						"name": "value",
+					},
+					Buckets: map[string]int{
+						"le-value": 0,
+					},
+				},
+			},
+			want: &HistogramVec{
+				{
+					Labels: map[string]string{
+						"name": "value",
+					},
+					Buckets: map[string]int{
+						"le-value": 1,
+					},
+				},
+			},
+		},
+		"convert multiple samples to bucket": {
+			samples: []*model.Sample{
+				{
+					Metric: model.Metric{"name": "value", "le": "le-value"},
+					Value:  1,
+				},
+				{
+					Metric: model.Metric{"name": "value", "le": "le-value"},
+					Value:  2,
+				},
+			},
+			inputHistogramVec: &HistogramVec{
+				{
+					Labels: map[string]string{
+						"name": "value",
+					},
+					Buckets: map[string]int{
+						"le-value": 0,
+					},
+				},
+			},
+			want: &HistogramVec{
+				{
+					Labels: map[string]string{
+						"name": "value",
+					},
+					Buckets: map[string]int{
+						"le-value": 3,
+					},
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		for _, sample := range test.samples {
+			ConvertSampleToBucket(sample, test.inputHistogramVec)
+		}
+		if !Equal(test.inputHistogramVec, test.want) {
```

Comment:
QQ, wouldn't using reflect.DeepEqual work here? If so, there is no need to create the Equal method.

Reply:
+1

Reply:
ha, it does indeed work. I created the Equal() method while I was working on the unit test to double check I was doing the correct thing. I replaced `Equal()` for `reflect.DeepEqual()` in the latest patch.